express-delete/server.ts

```diff
@@ -0,0 +1,60 @@
+import express from 'express';
+
+type Grade = {
+  id: number;
+  name: string;
+  course: string;
+  score: number;
+};
+
+type Grades = {
+  [key: number]: Grade;
+};
+
+const grades: Grades = {
+  12: {
+    id: 12,
+    name: 'Tim Berners-Lee',
+    course: 'HTML',
+    score: 95,
+  },
+  47: {
+    id: 47,
+    name: 'Brendan Eich',
+    course: 'JavaScript',
+    score: 100,
+  },
+  273: {
+    id: 273,
+    name: 'Forbes Lindsay',
+    course: 'JavaScript',
+    score: 92,
+  },
+};
+
+const app = express();
+
+app.get('/api/grades', (req, res) => {
+  const gradesArray: Grade[] = [];
+
+  for (const id in grades) {
+    gradesArray.push(grades[id]);
+  }
+
+  res.json(gradesArray);
+});
+
+app.delete('/api/grades/:id', (req, res) => {
+  const gradeId: number = Number(req.params.id);
```

Comment:
This is a case where TS can figure out that `gradeId` is a number without you telling it. I like to let TS do the work everywhere I can. I feel it makes the code easier to write, easier to read, and more flexible.